app/logo/[[...filename]]/page.tsx

```diff
@@ -0,0 +1,76 @@
+import { TinaClient } from "@/app/tina-client";
+import { TODAY } from "@/hooks/useFetchEvents";
+import { useSEO } from "@/hooks/useSeo";
+import client from "@/tina/client";
+import { Metadata } from "next";
+import LogoPage from ".";
+
+export const dynamicParams = false;
+
+type GenerateMetaDataProps = {
+  params: { filename: string[] };
+  searchParams: { [key: string]: string | string[] | undefined };
+};
+
+export async function generateMetadata({
+  params,
+}: GenerateMetaDataProps): Promise<Metadata> {
+  const tinaProps = await getData(params.filename);
+  const seo = tinaProps.props.seo;
+
+  if (seo && !seo.canonical) {
+    seo.canonical = `${tinaProps.props.header.url}logo/${params.filename}`;
```

Comment:
```suggestion
    seo.canonical = `${tinaProps.props.header.url}logo/${params.filename ?? ""}`;
```

this is when the canonical is empty and params.filename is undefined for index page